File: string/Solution14.java
class Solution14 {
    public String longestCommonPrefix(String[] strs) {
    	String result = "";
    	if(strs.length == 0){
    		return result;
    	}
    	for(int i = 0; i < strs[0].length(); i++){
    		char cur = strs[0].charAt(i);
    		for(String str: strs){
    			if(i >= str.length() || str.charAt(i) != cur){
    				return result;
    			}
    		}
    		result += cur;
    	}
    	return result;
    }
}

File: C++/Solution14.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution14 {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int size = strs.size();
        if(size == 0){
            return "";
        }
        string result = "";
        for(int i = 0; i < strs[0].length(); i++){
            char cur2 = strs[0][i];
            for(int index = 1; index < size; index++){
                string cur = strs[index];
                if(cur.length() <= i || cur[i] != cur2){
                    return result;
                }
            }
            result += cur2;
        }
        return result;
    }
};